### PR TITLE
Ifml

### PR DESCRIPTION
**Pull request by patricia-patricia**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "mxgraph": "3.7.2-2",
     "noty": "^3.1.2",
     "randomcolor": "^0.5.3",
-    "syncmeta-plugin": "git+https://github.com/rwth-acis/syncmeta-plugin.git",
+    "syncmeta-plugin": "git+https://github.com/rwth-acis/syncmeta-plugin.git#ifml",
```
Comment (pdolif) [inline package.json]: @patricia-patricia Can we merge the changes in syncmeta-plugin?

File: src/js/mapper/CAELiveMapper.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ import Noty from 'noty';
  * @memberof module:CAE-Mapper
  */
 function CAELiveMapper() {
-
+  console.log('CAELiveMapper.........................................................');
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: These log messages can be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 3]
```diff
@@ -85,36 +91,49 @@ function CAELiveMapper() {
...
-                    if(ytext)
-                        widgetAttr[attrib.name] = ytext;
-                    else mxLog.writeln('Widget node attribute ' + attrib.name + ' is not ytext????');
-                }*/
-                SyncMeta.setAttributeValue(widgetNodeId, 'width', Util.formatNumber(meta.getAttribute('width')));
-                SyncMeta.setAttributeValue(widgetNodeId, 'height', Util.formatNumber(meta.getAttribute('height')));
-            }
-
-            editor.graph.model.setAttribute('id', widgetNodeId);
+            // if (!widgetNodeId) {
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: Can these commented out lines be removed?

File: package.json (hunk 1 above)
Comment (patricia-patricia) [reply package.json]: Yes, we can remove the branch name.

File: src/js/mapper/CAELiveMapper.js (hunk 3 above)
Comment (patricia-patricia) [reply src/js/mapper/CAELiveMapper.js]: yes

Comment (patricia-patricia) [reply src/js/mapper/CAELiveMapper.js]: resolved

File: src/js/mapper/CAELiveMapper.js (hunk 2 above)
Comment (patricia-patricia) [reply src/js/mapper/CAELiveMapper.js]: resolved

File: package.json (hunk 1 above)
Comment (patricia-patricia) [reply package.json]: resolved

Comment (pdolif) [reply package.json]: Changes in syncmeta-plugin have been merged (https://github.com/rwth-acis/syncmeta-plugin/pull/3).

File: src/js/mapper/CAELiveMapper.js [hunk 4]
```diff
@@ -42,6 +41,8 @@ function CAELiveMapper() {
                     if (model.nodes.hasOwnProperty(key)) {
                         var node = model.nodes[key];
                         if (node.type === 'Widget') {
+                          console.log('widgetNodeId key');
+                          console.log(key);
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: May be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 5]
```diff
@@ -346,18 +326,33 @@ function CAELiveMapper() {
...
+                            // for (var key in model.nodes) {
+                            //     if (model.nodes.hasOwnProperty(key)) {
+                            //         var node = model.nodes[key];
+                            //         if (node.type === 'Widget') {
+                            //           console.log('widgetNodeId key');
+                            //           console.log(key);
+                            //             widgetNodeId = key;
+                            //         }
+                            //     }
+                            // }
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: Can this be removed?

File: src/js/mapper/CAELiveMapper.js [hunk 6]
```diff
@@ -387,12 +383,29 @@ function CAELiveMapper() {
...
+                        if (!widgetNodeId) {
+                            widgetNodeId = SyncMeta.createNode('Widget', 4500, 4500, 600, 600, 1, true);
+                            setTimeout(function () {
+                                var meta = editor.graph.model.getMeta();
+                                SyncMeta.setAttributeValue(widgetNodeId, 'width', Util.formatNumber(meta.getAttribute('width')));
+                                SyncMeta.setAttributeValue(widgetNodeId, 'height', Util.formatNumber(meta.getAttribute('height')));
+                                setTimeout(function () {
+                                    SyncMeta.applyLayout();
+                                    console.log('mmmm........');
+                                    console.log(model.nodes);
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: May be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 7]
```diff
@@ -402,32 +415,77 @@ function CAELiveMapper() {
...
-                                setTimeout(function () {
-                                    SyncMeta.applyLayout();
-                                    SyncMeta.createActivity('WireframeCreateEvent', cell.id, '<strong style="background: black; color: white;">CAE-WIREFRAME:</strong>\nCreated a new ' + cell.constructor.NAME, event.value);
-                                }, 1000);
-                            }, 1000);
+                            // if type create view component return
+                            // is syncmeta call always create timeout?
+                            console.log('type.....');
+                            console.log(cell.constructor.HTML_NODE_NAME); // div
+                            console.log(cell.value.getAttribute('uiType')); // divcontainer
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: May be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 8]
```diff
@@ -402,32 +415,77 @@ function CAELiveMapper() {
...
+
+                            if (cell.constructor.HTML_NODE_NAME === 'div') {
+                              console.log('cell2...');
+                              console.log(cell);
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: May be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 9]
```diff
@@ -402,32 +415,77 @@ function CAELiveMapper() {
...
+
+                                  setTimeout(function () {
+                                      SyncMeta.applyLayout();
+                                      SyncMeta.createActivity('WireframeCreateEvent', cell.id, '<strong style="background: black; color: white;">CAE-WIREFRAME:</strong>\nCreated a new ' + cell.constructor.NAME, event.value);
+                                  }, 1000);
+                              }, 1000);
+                            } else {
+                              console.log('parentt');
+                              console.log(editor.graph.model.getCell(event.value.parent));
+                              console.log(cell);
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: May be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 10]
```diff
@@ -402,32 +415,77 @@ function CAELiveMapper() {
...
+                              var parent = editor.graph.model.getCell(event.value.parent);
+                              console.log(cell.geometry.y);
+                              console.log(parent.geometry.y);
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: May be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 11]
```diff
@@ -402,32 +415,77 @@ function CAELiveMapper() {
...
+
+                              SyncMeta.createNode('View Component Part', offsetX, offsetY, 200, 50, 1, false, null, cell.id);
+                              setTimeout(function () {
+                                  var type = cell.constructor.HTML_NODE_NAME || cell.value.getAttribute('uiType');
+                                  SyncMeta.setAttributeValue(cell.id, 'type', type);
+                                  SyncMeta.setAttributeValue(cell.id, 'static', true);
+                                  SyncMeta.setAttributeValue(cell.id, 'id', type + '_' + Util.GUID().substr(0, 5));
+
+                                  console.log('widgetNodeId...........');
+                                  console.log(widgetNodeId);
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: May be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 12]
```diff
@@ -402,32 +415,77 @@ function CAELiveMapper() {
...
+
+                                  setTimeout(function () {
+                                      var parent = editor.graph.model.getCell(event.value.parent);
+                                      console.log('parent.......');
+                                      console.log(parent.id);
+                                      console.log('cell..........');
+                                      console.log(cell.id);
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: May be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 13]
```diff
@@ -402,32 +415,77 @@ function CAELiveMapper() {
...
+                                      if (parent && parent.id != '1') {
+                                        console.log('ok...........');
+                                          var edgeId = SyncMeta.createEdge('View Component to HTML', parent.id, cell.id);
+                                          console.log('end..........');
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: Logs may be removed.

File: src/js/mapper/CAELiveMapper.js [hunk 14]
```diff
@@ -453,11 +511,24 @@ function CAELiveMapper() {
...
                             var cell = editor.graph.model.getCell(id);
+                            var width = event.value.bounds[i].width + 100;
+                            var height = event.value.bounds[i].height + 100;
+
+                            console.log(typeof id);
+                            console.log(id);
+                            console.log(typeof width);
+                            console.log(width);
+                            console.log(typeof height);
+                            console.log(height);
```
Comment (pdolif) [inline src/js/mapper/CAELiveMapper.js]: May be removed.

**Review by pdolif — APPROVED** [review]

**Pull request merged**